contract/rust/src/lib.rs

```diff
@@ -1,11 +1,24 @@
+/*
+ * This is an example of a Rust smart contract that greets the account that calls it.
+ * Since account id's on NEAR human-readable, this contract might return "Hello near_friend"
+ * The two functions "set_greeting" and "welcome" change state and return state.
+ * Methods that access state are included in the "viewMethods" array in main.js
+ * Similarly, methods that change state are included in the "changeMethods" array.
+ *
+*/
+
+// To conserve gas, efficient serialization is achieved through Borsh (http://borsh.io/)
 use borsh::{BorshDeserialize, BorshSerialize};
 use near_bindgen::{env, near_bindgen};
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
+// Rust now allows for changing how memory is obtained, and wee_alloc is lightweight 
 #[global_allocator]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
+// Structs in Rust are similar to other languages, and may include impl keyword as shown below
```

Comment:
we could just contrast with `switch..case` statements which are very common in C-family languages.